### ISSUE TITLE
Bug: [no-var-requires, no-require-imports] support template literal strings for allow

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8186]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.3.3&fileType=.ts&code=MYewdgzgLgBAZiEMC8MBOBTAjgVwJaYAUABgHSkD0AJgIZQ2kBWE4xAlANxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDcBDaWtEQBHWJSGoMAbXDYciaNA7RIAGllysc7JF7x4HAO7owM7dsgAdSwDoAVsg5MAJJA3aAuu4gBfd198QHyA&tsconfig=&tokens=false

### Repro Code

```TypeScript
var packageData = require(`../package.json`);
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", { allow: ['\\.json$'] }],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

This should be allowed.

### Actual Result

```plaintext
Require statement not part of import statement.
```

Because the rule doesn't look at template literals inside call expressions, it still gets the complaint.

### Additional Info

Note that the production site won't have the `allow` option in the playground until the next release. I'm filing this as a fast follow from https://github.com/typescript-eslint/typescript-eslint/pull/7710#discussion_r1438366317
